### ISSUE TITLE
discrepancy in effect size CI for ggbetweenstats

### ISSUE THREAD

**Issue by IndrajeetPatil** [IndrajeetPatil/ggstatsplot#157]
Not sure why the values are so different for CI for omega-squared when the same function is being used in both contexts-
``` r

# creating the plot
set.seed(123)
ggstatsplot::ggbetweenstats(
  data = ggplot2::msleep,
  x = vore,
  y = brainwt,
  title = "mammalian sleep",
  caption = "From ggplot2 package",
  xlab = "vorarephilia",
  ylab = "brain weight",
  axes.range.restrict = TRUE,
  outlier.tagging = TRUE,
  outlier.label = name,
  conf.level = 0.99,
  k = 5,
  bf.message = TRUE,
  messages = FALSE
)
```

![](https://i.imgur.com/hwgXwc0.png)

``` r

# subtitle
set.seed(123)
ggstatsplot::subtitle_anova_parametric(
  data = ggplot2::msleep,
  x = vore,
  y = brainwt,
  k = 5,
  messages = FALSE,
  conf.level = 0.99
)
#> paste(NULL, italic("F"), "(", "3", ",", "24.04746", ") = ", "2.26528", 
#>     ", ", italic("p"), " = ", "0.10662", ", ", omega["p"]^2, 
#>     " = ", "0.00349", ", CI"["99%"], " [", "-0.10631", ", ", 
#>     "0.12739", "]", ", ", italic("n"), " = ", 51L)
```

<sup>Created on 2019-02-03 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                                             
#>  version  R Under development (unstable) (2018-11-30 r75724)
#>  os       Windows 10 x64                                    
#>  system   x86_64, mingw32                                   
#>  ui       RTerm                                             
#>  language (EN)                                              
#>  collate  English_United States.1252                        
#>  ctype    English_United States.1252                        
#>  tz       America/New_York                                  
#>  date     2019-02-03                                        
#> 
#> - Packages --------------------------------------------------------------
#>  package      * version     date       lib
#>  assertthat     0.2.0       2017-04-11 [1]
#>  backports      1.1.3       2018-12-14 [1]
#>  BayesFactor    0.9.12-4.2  2018-05-19 [1]
#>  bayesplot      1.6.0       2018-08-02 [1]
#>  boot           1.3-20      2017-08-06 [2]
#>  broom          0.5.1.9000  2019-01-20 [1]
#>  broom.mixed    0.2.3.9000  2019-01-23 [1]
#>  callr          3.1.1       2018-12-21 [1]
#>  cli            1.0.1.9000  2019-01-20 [1]
#>  cluster        2.0.7-1     2018-04-13 [2]
#>  coda           0.19-2      2018-10-08 [1]
#>  codetools      0.2-16      2018-12-24 [1]
#>  coin           1.2-2       2017-11-28 [1]
#>  colorspace     1.4-0       2019-01-13 [1]
#>  cowplot        0.9.4       2019-01-08 [1]
#>  crayon         1.3.4       2017-09-16 [1]
#>  curl           3.3         2019-01-10 [1]
#>  data.table     1.12.0      2019-01-13 [1]
#>  DEoptimR       1.0-8       2016-11-19 [1]
#>  desc           1.2.0       2019-01-21 [1]
#>  devtools       2.0.1.9000  2019-01-29 [1]
#>  digest         0.6.18      2018-10-10 [1]
#>  dplyr          0.8.0       2019-01-31 [1]
#>  effsize        0.7.4       2018-12-21 [1]
#>  emmeans        1.3.2       2019-01-22 [1]
#>  estimability   1.3         2018-02-11 [1]
#>  evaluate       0.12        2018-10-09 [1]
#>  exact2x2       1.6.3       2018-07-27 [1]
#>  exactci        1.3-3       2017-10-02 [1]
#>  fit.models     0.5-14      2017-04-06 [1]
#>  forcats        0.3.0       2018-02-19 [1]
#>  foreign        0.8-71      2018-07-20 [2]
#>  fs             1.2.6       2018-08-23 [1]
#>  generics       0.0.2       2019-01-20 [1]
#>  ggcorrplot     0.1.2.9000  2018-12-17 [1]
#>  ggExtra        0.8         2018-11-08 [1]
#>  ggplot2        3.1.0.9000  2019-01-31 [1]
#>  ggrepel        0.8.0       2018-05-09 [1]
#>  ggridges       0.5.1       2018-09-27 [1]
#>  ggsignif       0.4.0       2017-08-03 [1]
#>  ggstatsplot    0.0.8.9000  2019-02-03 [1]
#>  glmmTMB        0.2.3       2019-01-11 [1]
#>  glue           1.3.0       2018-07-17 [1]
#>  groupedstats   0.0.5.9000  2019-01-29 [1]
#>  gtable         0.2.0       2016-02-26 [1]
#>  gtools         3.8.1       2018-06-26 [1]
#>  haven          2.0.0       2018-11-22 [1]
#>  highr          0.7         2018-06-09 [1]
#>  hms            0.4.2       2018-03-10 [1]
#>  htmltools      0.3.6       2017-04-28 [1]
#>  httpuv         1.4.5.1     2018-12-18 [1]
#>  httr           1.4.0       2018-12-11 [1]
#>  jmv            0.9.6       2018-12-11 [1]
#>  jmvcore        0.9.5.2     2018-12-10 [1]
#>  knitr          1.21        2018-12-10 [1]
#>  labeling       0.3         2014-08-23 [1]
#>  later          0.7.5       2018-09-18 [1]
#>  lattice        0.20-38     2018-11-04 [2]
#>  lazyeval       0.2.1       2017-10-29 [1]
#>  lme4           1.1-19      2018-11-10 [1]
#>  magrittr       1.5         2014-11-22 [1]
#>  MASS           7.3-51.1    2018-11-01 [2]
#>  Matrix         1.2-15      2018-11-01 [1]
#>  MatrixModels   0.4-1       2015-08-22 [1]
#>  mc2d           0.1-18      2017-03-06 [1]
#>  memoise        1.1.0       2017-04-21 [1]
#>  metafor        2.0-0       2017-06-22 [1]
#>  mime           0.6         2018-10-05 [1]
#>  miniUI         0.1.1.1     2018-05-18 [1]
#>  minqa          1.2.4       2014-10-09 [1]
#>  mnormt         1.5-5       2016-10-15 [1]
#>  modelr         0.1.2       2018-05-11 [1]
#>  modeltools     0.2-22      2018-07-16 [1]
#>  multcomp       1.4-8       2017-11-08 [1]
#>  munsell        0.5.0       2018-06-12 [1]
#>  mvtnorm        1.0-8       2018-05-31 [1]
#>  nlme           3.1-137     2018-04-07 [2]
#>  nloptr         1.2.1       2018-10-03 [1]
#>  paletteer      0.1.0       2018-07-10 [1]
#>  pbapply        1.3-4       2018-01-10 [1]
#>  pcaPP          1.9-73      2018-01-14 [1]
#>  pillar         1.3.1       2018-12-15 [1]
#>  pkgbuild       1.0.2       2018-10-16 [1]
#>  pkgconfig      2.0.2       2018-08-16 [1]
#>  pkgload        1.0.2       2018-10-29 [1]
#>  plyr           1.8.4       2016-06-08 [1]
#>  prediction     0.3.6.2     2019-01-31 [1]
#>  prettyunits    1.0.2       2015-07-13 [1]
#>  processx       3.2.1       2018-12-05 [1]
#>  promises       1.0.1       2018-04-13 [1]
#>  ps             1.3.0       2018-12-21 [1]
#>  psych          1.8.12      2019-01-12 [1]
#>  purrr          0.3.0       2019-01-25 [1]
#>  purrrlyr       0.0.3       2018-05-29 [1]
#>  pwr            1.2-2       2018-03-03 [1]
#>  R6             2.3.0       2018-10-04 [1]
#>  Rcpp           1.0.0       2018-11-07 [1]
#>  remotes        2.0.2       2018-10-30 [1]
#>  reshape        0.8.8       2018-10-23 [1]
#>  reshape2       1.4.3       2017-12-11 [1]
#>  rjson          0.2.20      2018-06-08 [1]
#>  rlang          0.3.1       2019-01-08 [1]
#>  rmarkdown      1.11        2018-12-08 [1]
#>  robust         0.4-18      2017-04-27 [1]
#>  robustbase     0.93-3      2018-09-21 [1]
#>  rprojroot      1.3-2       2018-01-03 [1]
#>  rrcov          1.4-7       2018-11-15 [1]
#>  rstudioapi     0.9.0       2019-01-09 [1]
#>  sandwich       2.5-0       2018-08-17 [1]
#>  scales         1.0.0       2018-08-09 [1]
#>  sessioninfo    1.1.1       2018-11-05 [1]
#>  shiny          1.2.0       2018-11-02 [1]
#>  sjlabelled     1.0.16      2019-01-10 [1]
#>  sjmisc         2.7.7       2019-01-02 [1]
#>  sjstats        0.17.3.9000 2019-01-12 [1]
#>  skimr          2.0         2019-01-21 [1]
#>  snakecase      0.9.2       2018-08-14 [1]
#>  ssanv          1.1         2015-06-23 [1]
#>  stringdist     0.9.5.1     2018-06-08 [1]
#>  stringi        1.2.4       2018-07-20 [1]
#>  stringr        1.3.1       2018-05-10 [1]
#>  survival       2.43-3      2018-11-26 [2]
#>  testthat       2.0.1       2018-10-13 [1]
#>  TH.data        1.0-10      2019-01-21 [1]
#>  tibble         2.0.1       2019-01-12 [1]
#>  tidyr          0.8.2       2018-10-28 [1]
#>  tidyselect     0.2.5       2018-10-11 [1]
#>  TMB            1.7.15      2018-11-09 [1]
#>  usethis        1.4.0.9000  2019-01-29 [1]
#>  withr          2.1.2       2018-03-15 [1]
#>  WRS2           0.10-0      2018-06-15 [1]
#>  xfun           0.4         2018-10-23 [1]
#>  xml2           1.2.0       2018-01-24 [1]
#>  xtable         1.8-3       2018-08-29 [1]
#>  yaml           2.2.0       2018-07-25 [1]
#>  zoo            1.8-4       2018-09-19 [1]
#>  source                               
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  Github (tidymodels/broom@66411c4)    
#>  Github (bbolker/broom.mixed@fbe9597) 
#>  CRAN (R 3.6.0)                       
#>  Github (r-lib/cli@94e2fc5)           
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.2)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.0)                       
#>  Github (r-lib/desc@42b9578)          
#>  Github (r-lib/devtools@e4e57aa)      
#>  CRAN (R 3.5.1)                       
#>  Github (tidyverse/dplyr@67b9f4f)     
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  Github (r-lib/generics@d59e6b4)      
#>  local                                
#>  Github (daattali/ggExtra@f2a04df)    
#>  Github (tidyverse/ggplot2@54e773b)   
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  local                                
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  local                                
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (tidyverse/purrr@240f7b2)     
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  Github (strengejacke/sjstats@7b9a708)
#>  Github (ropenscilabs/skimr@d2b0b93)  
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.2)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  Github (r-lib/usethis@5a0680b)       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.1)                       
#>  CRAN (R 3.5.2)                       
#> 
#> [1] C:/Users/inp099/Documents/R/win-library/3.6
#> [2] C:/Program Files/R/R-devel/library
```

</details>

**Comment by IndrajeetPatil** [comment]
@ibecav I am stumped. Do you have any idea where this discrepancy might be coming from?

**Comment by ibecav** [comment]
No, nothing specific. I took a quick look when you first posted it but didn't see anything immediately.  My pure guess is that is has something to do with set.seed not carrying over since the actually omega values are identical and it's just the boot-strapped CI's that are slightly different.

But that's speculation at this point.

I'll likely look Friday.  I'm trying to wrap some work I have been doing (ironically) on cohen and hedges for ggbetweenstats.  Started with my own work and then drifted over.  If I finish up you should be able to drop the dependency for `effsize` completely and get the same answers .

**Comment by IndrajeetPatil** [comment]
Okay, thanks. 

That's exciting! Are you considering including the new functions in `ggstatsplot` itself or turning those functions into a lighter package that `ggstatsplot` can then add as one of its dependencies? 

Apropos this conversation, I've completely changed my thought process about dependencies after watching this talk-
https://resources.rstudio.com/rstudio-conf-2019/it-depends-a-dialog-about-dependencies 

I think you'll enjoy it as well.

**Comment by ibecav** [comment]
Nice presentation and I would never undertake to replace a full blown package like `psych` that is well maintained.  

My thinking on `cohen.d` is to just make it a function within helpers_ggbetween_subtitles so it's a simple dropin replacement for `effsize`

**Comment by IndrajeetPatil** [comment]
Okay, but just so you know, `effsize` is the lightest dependency we have. It literally depends on **0** other packages, not counting the base packages (https://github.com/mtorchiano/effsize/blob/master/NAMESPACE). So removing this dependency is not going to make much of a difference from the user's or developer's perspective.

**Comment by ibecav** [comment]
Well it's been interesting.  Some background.  `effsize::cohend` has some bad math in it.  If you look at the help page the author was applying the two sample formula he shows there to other cases generating bad data. (see https://muse.jhu.edu/chapter/284934) I also tried to use the `psych::cohend` but it had an error as well which I reported to Bill Revelle he just responded this morning...

"Definitely a bug.

I had redefined d to be cohen’s d and then divided that again.
(The code read well, it was just wrong!).
This will be corrected in the next release.

For now, you can get the working version from my lab sever.  1.9.2
install.packages("psych",repos="https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fpersonality-project.org%2Fr&amp;data=02%7C01%7Cchuck.powell%40trincoll.edu%7C342eaee482504cf77f8e08d690a15c13%7Ca6cda06a52d94672ae1ff6f9c9f14e37%7C0%7C0%7C636855420081375738&amp;sdata=MKMp%2F6jUHqX7MVAWBBzEuSIfZg%2FZVxCdIVUBdCSgfhE%3D&amp;reserved=0",type=“source")

The eventual CRAN release will be a few months from now.

Bill"

Meantime I rewrote the code in lsr::cohensD to get readable and accurate code.

As noted in the video you sent not my first choice and lots of joy but it's done the PR is almost ready I just have to chase down one last bug in the case where it's a paired case but we're accepting the data in long format.

**Comment by IndrajeetPatil** [comment]
Thanks for all the work, Chuck!

I will work on adding tests for the new function.

In the meantime, do you now want to work on the effect size for the one-sample and two-sample non-parametric tests?

Although the deadline is the 19th, I plan to submit `0.0.9` to `CRAN` on the 15th Feb., just to be safe.

**Comment by ibecav** [comment]
LOL, well I'd like to say it was a pleasure but that would be a lie.  There were moments I wanted to throw my computer in the trash.  But I think it was a good thing to do.  i don't actually like to write my own code but in this case every other path seemed to lead to someone else's bugs.

Yes I'll get to those next.  No promise on when exactly i'm way behind on my day job (I signed up to be on an accreditation team visit to another school ...)

I did learn a lot though and I'll get to the other bits.  My apologies for being more of a statistician than a programmer.

**Comment by IndrajeetPatil** [comment]
Yeah, I can totally see how much work has gone into writing this function! (And I am very much familiar with the urge to trash the computer when you can't get something to work :) )

No worries! If not in this version, we can probably postpone this to the next release. But, I'm just worried that users might not be happy with the constant changes in effect sizes across versions. So wanted to have this one in the current version.

By the way, since we are exporting the new function, how do you feel about calling it `effsize_t_parametric` rather than `effect_t_parametric`? 
The former will be easier to remember, IMO.

**Comment by ibecav** [comment]
No real opinion on function name, feel free if you think it will help.  I actually didn't pay much attention to the fact it was exported.

**Comment by IndrajeetPatil** [comment]
Cool. 

By the way, how can use the formula interface for running one sample t-test with this function?

**Comment by ibecav** [comment]
Working on doco for a couple of minutes be fore you change the name

``` r
library(ggstatsplot)
moon <- c(1.73, 1.06, 2.03, 1.40, 0.95, 1.13, 1.41, 1.73, 1.63, 1.56)
moondata <- data.frame(moon)
effect_t_parametric(~ moon, data = moondata, mu = 1)
#> # A tibble: 1 x 4
#>   method     estimate conf.low conf.high
#>   <chr>         <dbl>    <dbl>     <dbl>
#> 1 Hedges's g     1.23    0.491      2.34
```

<sup>Created on 2019-02-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by IndrajeetPatil** [comment]
Ah, I have already done a number of changes. Do you mind waiting on me to push them and then you can change the documentation?

**Comment by ibecav** [comment]
Tell you what I'll just send you the diffs...

**Comment by ibecav** [comment]
This ...

#' @details
#'
#' This function is a rewrite of functionality provided in `lsr::cohensD` and
#' `effsize::cohen.d`
#'
#'
#' References:
#' \itemize{
#'  \item Cooper, Harris, Hedges, Larry V., Valentine, Jeffrey C.,
#'    *The Handbook of Research Synthesis and Meta-Analysis*, 2009
#'  \item Cumming, G., Finch, S.,
#'    *A Primer On The Understanding, Use, And Calculation Of Confidence Intervals That Are Based On Central And Noncentral Distributions*,
#'    Educational and Psychological Measurement, Vol. 61 No. 4, August 2001 532-574
#'  \item Cohen, J. (1988). *Statistical power analysis for the behavioral sciences (2nd ed.).* Hillsdale, NJ: Lawrence Erlbaum Associates.
#'  \item David C. Howell (2010). *Confidence Intervals on Effect Size.* retrieved from https://www.uvm.edu/~dhowell/methods7/Supplements/Confidence%20Intervals%20on%20Effect%20Size.pdf.
#'  }
#'

**Comment by IndrajeetPatil** [comment]
Done!

**Comment by IndrajeetPatil** [comment]
While adding tests for the new effsize function, I noticed that 95% CI bounds are incorrect when the effect size is-
- Cohen's _d_ 
- `paired = TRUE` 
- `centrality = TRUE`

``` r
# creating a new dataframe with a variable that has dropped factor level
mtcars_short <- dplyr::filter(.data = mtcars, cyl != "4")

# significance with stats package
stats::t.test(
  formula = wt ~ cyl,
  data = mtcars_short,
  paired = FALSE,
  conf.level = 0.99
)

#>  Welch Two Sample t-test
#> 
#> data:  wt by cyl
#> t = -3.6212, df = 18.991, p-value = 0.00182
#> alternative hypothesis: true difference in means is not equal to 0
#> 99 percent confidence interval:
#>  -1.5789832 -0.1851597

# confidence interval contains 0, although the effect is significant
ggstatsplot::effsize_t_parametric(
  formula = wt ~ cyl,
  data = mtcars_short,
  paired = FALSE,
  hedges.correction = FALSE,
  conf.level = 0.99,
  noncentral = FALSE
)

#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <lgl> 
#> 1 Cohen~    -1.34    -2.79     0.112       0.99 two.sided   FALSE 
#> # ... with 2 more variables: noncentral <lgl>, var.equal <lgl>

# correct output with noncentral CIs
ggstatsplot::effsize_t_parametric(
  formula = wt ~ cyl,
  data = mtcars_short,
  paired = FALSE,
  hedges.correction = FALSE,
  conf.level = 0.99,
  noncentral = TRUE
)

#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <lgl> 
#> 1 Cohen~    -1.34    -2.64   -0.0112       0.99 two.sided   FALSE 
#> # ... with 2 more variables: noncentral <lgl>, var.equal <lgl>
```

<sup>Created on 2019-02-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by IndrajeetPatil** [comment]
There is also a bug in the one-sample t-test effect sizes, but only for the following combination. All other combinations look okay.

``` r
# significant effect
stats::t.test(x = dplyr::starwars$height,
              mu = 100,
              na.action = na.omit,
              conf.level = 0.99)

#>  One Sample t-test
#> 
#> data:  dplyr::starwars$height
#> t = 19.247, df = 80, p-value < 2.2e-16
#> alternative hypothesis: true mean is not equal to 100
#> 99 percent confidence interval:
#>  164.1638 184.5523
#> sample estimates:
#> mean of x 
#>   174.358

# confidence interval contains 0
ggstatsplot::effsize_t_parametric(
  formula = ~ height,
  data = dplyr::starwars,
  hedges.correction = TRUE,
  mu = 100,
  conf.level = 0.99,
  noncentral = FALSE
)

#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <chr> 
#> 1 Hedge~     2.12    -13.4      17.7       0.99 two.sided   <NA>  
#> # ... with 2 more variables: noncentral <lgl>, var.equal <lgl>
```

<sup>Created on 2019-02-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by ibecav** [comment]
Two things.

1. I'm a little confused by what you're worried is a bug here?  You're not worried because our confidence for d or g crosses zero are you?  That is entirely plausible no matter how "significant" the t test says the mean difference is.  Maybe I'm jusy missing what you're trying to tell me though so pease explain if i'm missing your point.

2.  One of the things I learned in my deep dive into effect sizes and confidence intervals is that ncp should definitely be the default.  More accuracy and there is zero reason to assume that the variance of d or g is central around 0.  A couple of reasonable articles on topic.  https://journals.sagepub.com/doi/pdf/10.1177/0013164401614002 and https://www.uvm.edu/~dhowell/methods7/Supplements/Confidence%20Intervals%20on%20Effect%20Size.pdf

**Comment by IndrajeetPatil** [comment]
1. Pardon my ignorance here since I am not well-versed in math behind effect size computations. 
Yes, that was partly my concern. I've learned from other threads that this can happen across bootstrapped effect size intervals or _p_-values versus traditional _p_-value (e.g., https://stats.stackexchange.com/questions/302928/disagreement-between-p-values-and-confidence-intervals), but was not sure if this happens even in non-bootstrapped contexts.

But my bigger concern is that the confidence intervals from central and non-central methods for the same effect size come to diametrically opposite conclusion. (This is especially jarring for the one sample test)

two-sample -

![image](https://user-images.githubusercontent.com/11330453/52718611-e289c380-2f71-11e9-914f-9b88b9990bfa.png)

one-sample - 

![image](https://user-images.githubusercontent.com/11330453/52718626-eae1fe80-2f71-11e9-820f-020223af7bda.png)

2. I am completely open to this suggestion and we can definitely change the default before the next release.

**Comment by ibecav** [comment]
Okay I broke out my high school algebra skills and came up with a better estimate for Sigmad the variance around d.  It seems to yield a much more intuitively correct answer and since it is just n algebraic reformulation of the two sample case I'm confident.  PR follows 

``` r
library(ggstatsplot)

ggstatsplot::effsize_t_parametric(formula = ~ height, 
                                  dplyr::starwars, 
                                  hedges.correction = TRUE, 
                                  mu = 100, 
                                  conf.level = .99, 
                                  noncentral = TRUE)
#> Warning in stats::pt(q = tvalue, df = dfvalue, ncp = x): full precision may
#> not have been achieved in 'pnt{final}'

#> Warning in stats::pt(q = tvalue, df = dfvalue, ncp = x): full precision may
#> not have been achieved in 'pnt{final}'
#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <chr> 
#> 1 Hedge…     2.12     1.63      2.68       0.99 two.sided   <NA>  
#> # … with 2 more variables: noncentral <lgl>, var.equal <lgl>
ggstatsplot::effsize_t_parametric(formula = ~ height, 
                                  dplyr::starwars, 
                                  hedges.correction = TRUE, 
                                  mu = 100, 
                                  conf.level = .99, 
                                  noncentral = FALSE)
#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <chr> 
#> 1 Hedge…     2.12     1.40      2.87       0.99 two.sided   <NA>  
#> # … with 2 more variables: noncentral <lgl>, var.equal <lgl>
```

<sup>Created on 2019-02-13 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by IndrajeetPatil** [comment]
This will only address the Hedge's _g_, right?

There is a similar issue with Cohen's _d_ as well:
``` r
ggstatsplot::effsize_t_parametric(formula = ~ height, 
                                  dplyr::starwars, 
                                  hedges.correction = FALSE, 
                                  mu = 100, 
                                  conf.level = 0.99, 
                                  noncentral = FALSE)

#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <chr> 
#> 1 Cohen~     2.14    -13.4      17.7       0.99 two.sided   <NA>  
#> # ... with 2 more variables: noncentral <lgl>, var.equal <lgl>
```

These confidence intervals are wildely different from what one would expect (e.g., here is output from `JASP`)-

![image](https://user-images.githubusercontent.com/11330453/52749604-ac6f3280-2fb7-11e9-9a67-0b5dbfc796be.png)

These results are more in line with when `noncentral = TRUE`-
``` r
ggstatsplot::effsize_t_parametric(formula = ~ height, 
                                  dplyr::starwars, 
                                  hedges.correction = FALSE, 
                                  mu = 100, 
                                  conf.level = 0.99, 
                                  noncentral = TRUE)

#> Warning in stats::pt(q = tvalue, df = dfvalue, ncp = x): full precision may
#> not have been achieved in 'pnt{final}'

#> Warning in stats::pt(q = tvalue, df = dfvalue, ncp = x): full precision may
#> not have been achieved in 'pnt{final}'

#> # A tibble: 1 x 9
#>   method estimate conf.low conf.high conf.level alternative paired
#>   <chr>     <dbl>    <dbl>     <dbl>      <dbl> <chr>       <chr> 
#> 1 Cohen~     2.14     1.63      2.68       0.99 two.sided   <NA>  
#> # ... with 2 more variables: noncentral <lgl>, var.equal <lgl>
```

**Comment by IndrajeetPatil** [comment]
Can you post a reprex with the new version of the function (in PR) with the example I just used?

**Comment by ibecav** [comment]
Nope applies to both

**Comment by ibecav** [comment]
So you want the reprex with the PR instead of here with the issue?

**Comment by IndrajeetPatil** [comment]
No, no. Both of the examples you showed were with Hedge's _g_ and I wanted to see what Cohen's _d_ looks like. But I see from changed tests that the patch fixes both of these effect sizes and not just Hedge's _g_. So an additional reprex here for Cohen's _d_ is unnecessary.

![image](https://user-images.githubusercontent.com/11330453/52751364-3372d980-2fbd-11e9-9014-f57de343c559.png)

**Comment by ibecav** [comment]
Ah, yes ncp runs after d or g is calculated so it is applied to whichever is in play
